### PR TITLE
autoloop: include peer rules for inflight/budget limits

### PR DESCRIPTION
**Pull request by carlaKC**
If we ran into a constraint while dispatching new autoloops (hit our in-flight limit or budget), we were not setting a disqualified reason, which would mean that this reasoning isn't surfaced to the user. This PR adds a test to demonstrate the bug and addresses it.

#### Pull Request Checklist
- [x] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: liquidity/interface.go [hunk 1]
```diff
@@ -69,3 +75,24 @@ func (l *loopOutSwapSuggestion) channels() []lnwire.ShortChannelID {
 
 	return channels
 }
+
+// peers returns the set of peers that the loop out swap is restricted to.
+func (l *loopOutSwapSuggestion) peers(knownChans map[uint64]route.Vertex) []route.Vertex {
+	peers := make(map[route.Vertex]struct{})
```
Comment (bhandras) [inline liquidity/interface.go:81]: nit: `make(map[route.Vertex]struct{}, len(l.OutgoingChanSet))`

File: liquidity/interface.go [hunk 2]
```diff
@@ -69,3 +75,24 @@ func (l *loopOutSwapSuggestion) channels() []lnwire.ShortChannelID {
 
 	return channels
 }
+
+// peers returns the set of peers that the loop out swap is restricted to.
+func (l *loopOutSwapSuggestion) peers(knownChans map[uint64]route.Vertex) []route.Vertex {
```
Comment (bhandras) [inline liquidity/interface.go:80]: nit: could break the line

**Review by bhandras — APPROVED** [review]
LGTM :palm_tree:

**Review by guggero — APPROVED** [review]
LGTM :tada:

**Comment by carlaKC** [comment]
Addressed nits and added a compile time assertion that we satisfy our interface - [diff](https://github.com/lightninglabs/loop/compare/d265bc9bee654b0f69c986001e2d7d323e4c33ed..cc10d5b207568f5d8b0dabd5bb5e61913f38ac1a).

**Pull request merged**